recipes/zserio/all/conanfile.py

```diff
@@ -0,0 +1,79 @@
+import os
+import shutil
+
+from conan import ConanFile
+from conan.tools.files import copy, get, collect_libs
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=2.0.0"
+
+class ZserioConanFile(ConanFile):
+    name = "zserio"
+    description = "Zserio C++ Runtime Library"
+    license = "BSD-3 Clause"
+    url = "https://github.com/conan-io/conan-center-index/"
+    homepage = "https://zserio.org"
+    topics = ("zserio", "C++", "serialization")
+    package_type = "static-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if Version(self.version) < "2.13.0":
+            raise ConanInvalidConfiguration("Minimum available version is 2.13.0")
+
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version]["runtime"],
+            pattern="runtime_libs/cpp/*", strip_root=True)
+        shutil.move("cpp/zserio", ".")
+        shutil.move("cpp/CMakeLists.txt", ".")
+        shutil.rmtree("cpp")
```

Comment:
```suggestion
```
Please, move files during build step, source should be immutable.

Reply:
This is just an additional step to get **only** zserio c++ runtime sources. Something like additional functionality to the `pattern` argument of the `get` function. The zserio release contains also sources for Java and Python runtime and I just wanted to keep only C++ sources in a proper source folder. Is it still wrong?

If I will have to move it elsewhere, I would probably do it before `generate` to let cmake find the `CMakeLists.txt`.

Reply:
We don't filter neither modify the download package, the source folder should be immutable, just like downloading it without changes. I see the package is only 14MB, so it's not big enough. Please, discard those lines, including the `pattern` from `get`. If you need to move something, please, do it in the build, when the source code is copied in the build folder yet. You can use `self.source_folder` as your root.

Reply:
ok, done, using `build_script_folder` now:
https://github.com/ndsev/zserio-conan-center-index/blob/569d8ce57111559a1bdc82bf98c2f80116c93bec/recipes/zserio/all/conanfile.py#L73